src/test/java/com/iota/iri/controllers/TipsViewModelTest.java

```diff
@@ -136,4 +157,67 @@ public void tipsEmtpyAfterClear() throws Exception {
         assertEquals(0, tipsVM.size());
     } 
 
-}
\ No newline at end of file
+    @Test
+    public void solidTipsPopulatedWhenEmpty() throws Exception {
+        final int tipCount = 10;
+        Set<Hash> tips = new HashSet<>();
+
+        for (int i = 0; i < tipCount; i++) {
+            TransactionViewModel newTip = new TransactionViewModel(
+                getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+            newTip.updateSolid(true);
+
+            newTip.store(tangle);
+            tips.add(newTip.getHash());
+        }
+
+        TipsViewModel tipsVM = new TipsViewModel(tangle);
+
+        Hash tip = tipsVM.getRandomSolidTipHash();
+
+        assertNotNull(tip);
+        assertTrue(tips.contains(tip));
+    }
+
+    @Test
+    public void nonSolidTipsAreIgnoredWhenPopulatingSolidTips() throws Exception {
+        TransactionViewModel solidTip = new TransactionViewModel(
+            getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+        TransactionViewModel nonSolidTip = new TransactionViewModel(
+            getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+
+        solidTip.updateSolid(true);
+
+        solidTip.store(tangle);
+        nonSolidTip.store(tangle);
+
+        TipsViewModel tipsVM = new TipsViewModel(tangle);
+
+        Hash tip = tipsVM.getRandomSolidTipHash();
+
+        assertEquals(solidTip.getHash(), tip);
```

Comment:
would be better to explicitly show that `nonSolidTip` is not in the tips list.
this test could theoretically pass/fail wp 1/2 even if there is a bug and solid are not ignored.